scrapy/core/http2/stream.py

```diff
@@ -185,14 +185,32 @@ def _get_request_headers(self) -> List[Tuple[str, str]]:
         if url.query:
             path += '?' + url.query
 
+        # This pseudo-header field MUST NOT be empty for "http" or "https"
+        # URIs; "http" or "https" URIs that do not contain a path component
+        # MUST include a value of '/'. The exception to this rule is an
+        # OPTIONS request for an "http" or "https" URI that does not include
+        # a path component; these MUST include a ":path" pseudo-header field
+        # with a value of '*' (refer RFC 7540 - Section 8.1.2.3)
+        if not path:
+            if self._request.method == 'OPTIONS':
+                path = path or '*'
+            else:
+                path = path or '/'
```

Comment:
:lipstick: (`if not path` allows simplifying inner code)

```suggestion
        if not path:
            path = '*' if self._request.method == 'OPTIONS' else '/'
```